### PR TITLE
feat: add XDeFi Adapter (fixed tests)

### LINKED ISSUE #407: XDEFI wallet support hdwallet
taking over @loxator's PR as he's out this week

### PR DESCRIPTION
**Pull request by loxator**
Migrated from #405 , fixes #407

### CODE REVIEW

**Comment by loxator** [comment]
@mrnerdhair Build works fine now with the tests passing :partying_face:

**Review by cjthompson — APPROVED** [review]

File: packages/hdwallet-xdefi/src/xdefi.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  public hasOnDeviceDisplay(): boolean {
+    return false;
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return false;
+  }
+
+  public hasNativeShapeShift(): boolean {
+    // It doesn't... yet?
```
Comment (mrnerdhair) [inline packages/hdwallet-xdefi/src/xdefi.ts:232]: (It never will, this "native shapeshift" thing has to do with dead legacy stuff from when ShapeShift was an exchange)
```suggestion
```

**Review by mrnerdhair — APPROVED** [review]
This is great, high-quality stuff, and thanks for hanging in there with us through this process. I'd be fine with getting this merged as-is, but can you verify that `integration/src/wallets/mocks/@xdefi/provider.ts` is intended to be present but empty? (If so, maybe a comment about why would make sense?)

**Comment by loxator** [comment]
> integration/src/wallets/mocks/@xdefi/provider.ts

Good catch, I was trying something out and forgot to remove it.

**Pull request merged**